### PR TITLE
Decompress response body string if it was compressed

### PR DESCRIPTION
**Pull request by rmstar**
- getResponseBodyAsString() now returns uncompressed data if response was compressed using gzip or deflate
-  Added new API getUncompressedResponseBody() that returns uncompressed response as bytes if it was response was compressed
- Upgraded checkstyle version (now using same version as cdap repo) and fixed checkstyle issues

### CODE REVIEW

File: common-cli/pom.xml [hunk 1]
```diff
@@ -22,7 +22,7 @@ the License.
   <parent>
     <artifactId>common</artifactId>
     <groupId>io.cdap.common</groupId>
-    <version>0.13.0-SNAPSHOT</version>
+    <version>0.13.1</version>
```
Comment (tivv) [inline common-cli/pom.xml]: Please put "-SNAPSHOT until after released

File: common-http/src/main/java/io/cdap/common/http/HttpResponse.java [hunk 2]
```diff
@@ -86,12 +90,44 @@ public byte[] getResponseBody() {
...
+      if ("deflate".equalsIgnoreCase(encoding)) {
+        try (InputStream is = new DeflaterInputStream(new ByteArrayInputStream(responseBody))) {
+          return ByteStreams.toByteArray(is);
+        }
+      }
+    } catch (IOException e) {
+      throw new IllegalStateException(String.format("Failed to decompress %s encoded response body", encoding), e);
+    }
+
+    throw new IllegalArgumentException("Unsupported content encoding " + encoding);
```
Comment (tivv) [inline common-http/src/main/java/io/cdap/common/http/HttpResponse.java]: Use `IllegalStateException` as it's not an argument

**Review by tivv — COMMENTED** [review]
Also please add a unit test

File: common-cli/pom.xml (hunk 1 above)
Comment (rmstar) [reply common-cli/pom.xml]: reverted

File: common-http/src/main/java/io/cdap/common/http/HttpResponse.java (hunk 2 above)
Comment (rmstar) [reply common-http/src/main/java/io/cdap/common/http/HttpResponse.java]: fixed

**Comment by rmstar** [comment]
> Also please add a unit test

added test

**Review by tivv — APPROVED** [review]

**Pull request merged**